lib/wait.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Maze
+  # Allows repeated attempts at something, until it is successful or the timeout
+  # is exceed
+  class Wait
+    # @param interval [Numeric] Optional. The time to sleep between attempts
+    # @param timeout [Numeric] The amount of time to spend on attempts before giving up
+    def initialize(interval: 0.1, timeout:)
+      @interval = interval
+      @max_attempts = interval / timeout
+    end
+
+    # Wait until the given block succeeds (returns a truthy value) or the
+    # timeout is exceeded
+    #
+    # @return [Object] The last value returned by the block
+    def until(&block)
+      success = false
+      attempts = 0
```

Comment:
Need to increment `attempts`

Reply:
Yeah this class was just busted, fixed in https://github.com/bugsnag/maze-runner/pull/183/commits/40ec240405e39715ca49bd3eccca69d3c89f9e51